pkg/protocol/engine/engine.go

```diff
@@ -127,8 +128,9 @@ func New(
 			LatestCommitment: reactive.NewVariable[*model.Commitment](),
 			Workers:          workers,
 
-			optsSnapshotPath:  "snapshot.bin",
-			optsSnapshotDepth: 5,
+			optsSnapshotPath:    "snapshot.bin",
+			optsSnapshotDepth:   5,
+			optsCheckCommitment: true,
```

Comment:
Add an option so that this can be modified. Take snapshot path as a reference.

Reply:
Made this